sdk/nodejs/runtime/closure/createClosure.ts

```diff
@@ -19,7 +19,8 @@ import { ResourceError } from "../../errors";
 import { Input, isSecretOutput, Output } from "../../output";
 import * as resource from "../../resource";
 import { hasTrueBooleanMember } from "../../utils";
-import { CapturedPropertyChain, CapturedPropertyInfo, CapturedVariableMap, parseFunction } from "./parseFunction";
+import { CapturedPropertyChain, CapturedPropertyInfo, CapturedVariableMap } from "./parseFunction";
+import * as parseFunctionModule from "./parseFunction";
```

Comment:
Your importing here and calling require below? Doesn't this import trigger an un-conditional require?
Same with typescript in the other files.

Reply:
It doesn't. Same technique as in https://github.com/pulumi/pulumi/pull/10185 

I was learning from https://www.typescriptlang.org/docs/handbook/modules.html#optional-module-loading-and-other-advanced-loading-scenarios and some experimentation.

When an import is "compile time only" like importing Interface types, it gets elided during translation to JavaScript.

So the pattern is:

```
import * as foo from "foo";  // compile-time only import IFF no foo.func() calls are made; erased from JS code

....
    if (cond) {
         var actualFoo: typeof foo = require("foo");  // require() gets compiled to actual JS code
         // since the typeof declaration above, actualFoo.callFunc() gets type-checked properly, it's not <any>
    }


```